File: rest-api/src/main/java/com/kt/kol/api/post/model/Post.java
package com.kt.kol.api.post.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("posts")
public class Post {
    @Id
    private Long id;

    private String title;
    private String content;
    private Long userId;
    private String status;
    private Integer viewCount;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: rest-api/src/main/kotlin/com/kt/kol/api/post/model/Post.kt
package com.kt.kol.api.post.model

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime

@Table("posts")
data class Post(
    @Id
    val id: Long? = null,
    var title: String? = null,
    var content: String? = null,
    var userId: Long? = null,
    var status: String? = null,
    var viewCount: Int? = null,
    var createdAt: LocalDateTime? = null,
    var updatedAt: LocalDateTime? = null
)